### PR TITLE
Catch a bunch of errors everywhere:

### PR DESCRIPTION
**Pull request by naved001**
This PR is pretty much copy-pasta from @chemistry-sourabh's #111 PR
since dealing with the merge conflicts was nasty and it made some changes
which I now feel are not necessary.

This has some fixes for database issues, ceph image issues, and tgt targets. And it's weird that we have to catch integrity errors, the code should be such that it's never even raised (I'll look fix that within this PR or maybe a different one).

I will add some integration tests too.

@apoorvemohan @mihirborkar @izhmash

### CODE REVIEW

File: ims/einstein/iscsi/tgt.py [hunk 1]
```diff
@@ -125,14 +125,19 @@ def remove_target(self, target_name):
         try:
             targets = self.list_targets()
             if target_name in targets:
-                os.remove(os.path.join(self.TGT_ISCSI_CONFIG,
-                                       target_name + ".conf"))
                 command = "tgt-admin -f --delete {0}".format(target_name)
                 output = shell.call(command, sudo=True)
                 logger.debug("Output = %s", output)
+                os.remove(os.path.join(self.TGT_ISCSI_CONFIG,
```
Comment (ianballou) [inline ims/einstein/iscsi/tgt.py:131]: Why did you have to move this command?

**Review by ianballou — APPROVED** [review]
I think this looks cool

**Comment by naved001** [comment]
Thanks for the review, but I think this will require more work once #174 is merged since that restructures a lot of things.

File: ims/einstein/ceph.py [hunk 2]
```diff
@@ -145,15 +149,15 @@ def list_children(self, img_id, parent_snap):
...
         except rbd.ImageNotFound:
-            raise file_system_exceptions.ImageNotFoundException(img_id)
+            logger.info("%s image is not found" % img_id)
         # Don't know how to raise this
         except rbd.ImageBusy:
             raise file_system_exceptions.ImageBusyException(img_id)
         # Forgot to test this
         except rbd.ImageHasSnapshots:
             raise file_system_exceptions.ImageHasSnapshotException(img_id)
+        return true
```
Comment (jeremyfreudberg) [inline ims/einstein/ceph.py:160]: welll, that's something

File: ims/einstein/iscsi/tgt.py [hunk 3]
```diff
@@ -125,14 +125,19 @@ def remove_target(self, target_name):
...
+                                       target_name + ".conf"))
             else:
-                raise iscsi_exceptions.TargetDoesntExistException()
-        except (IOError, OSError) as e:
+                logger.info("%s target doesnt exist" % target_name)
+        except OSError as e:
+            if "[Errno 2] No such file or directory" in str(e):
```
Comment (jeremyfreudberg) [inline ims/einstein/iscsi/tgt.py:136]: you should probably compare the value of `e.errno` to `2`, or for bonus points, compare it to `os.errno.ENOENT`

**Review by jeremyfreudberg — CHANGES_REQUESTED** [review]

File: ims/einstein/ceph.py (hunk 2 above)
Comment (naved001) [reply ims/einstein/ceph.py:160]: oops

**Comment by ianballou** [comment]
How's this code looking now with the recent changes? Is everything still relevant?

**Pull request closed without merging**